index.js

```diff
@@ -124,42 +124,41 @@ module.exports = function (host_or_multiaddr, port) {
   function getFileStream (files) {
     if (!files) return null
     if (!Array.isArray(files)) files = [files]
+    if (files.length === 0) throw new Error('no files given')
 
-    var file
+    var fstream = new stream.PassThrough({objectMode: true })
 
     for (var i = 0; i < files.length; i++) {
-
-      file = files[i]
+      var file = files[i]
 
       if (typeof file === 'string') {
-        file = new File({
-          cwd: path.dirname(file),
-          base: path.dirname(file),
-          path: file,
-          contents: fs.createReadStream(file)
-        })
+        vinylfs.src(file).pipe(fstream)
+
       } else if (Buffer.isBuffer(file)) {
-        file = new File({
+        fstream.push(new File({
           cwd: '/',
           base: '/',
           path: '/',
           contents: streamifier.createReadStream(file)
-        })
+        }))
+
       } else if (file instanceof stream.Stream) {
-        file = new File({
+        fstream.push(new File({
           cwd: '/',
           base: '/',
           path: '/',
           contents: file
-        })
-      } else if (!file instanceof File) {
-        return null
```

Comment:
Looked at this again, definitely wrong, i was parsing the for as a forEach, so that the return would just keep the loop going. i'll look at it.